climada/hazard/centroids/centr.py

```diff
@@ -491,52 +491,83 @@ def clear(self):
         self.__init__()
 
     def append(self, centr):
-        """Append raster or points.
+        """Append centroids points.
+
+        If centr is raster it is converted to points first using
+        Centroids.set_meta_to_lat_lon. Raster information will be lost.
+
+        Note: self is modified in-place, and meta is set to {}.
 
         Parameters
         ----------
         centr : Centroids
-            If it's a raster, it needs to have the same `meta` attribute.
-            If it's of non-raster form, it's geometry needs to have the same CRS.
+            Centroids to append. The centroids need to have the same CRS.
         """
-        if self.meta and centr.meta:
-            LOGGER.debug('Appending raster')
-            if centr.meta['crs'] != self.meta['crs']:
-                raise ValueError('Different CRS not accepted.')
-            if self.meta['transform'][0] != centr.meta['transform'][0] \
-               or self.meta['transform'][4] != centr.meta['transform'][4]:
-                raise ValueError('Different raster resolutions.')
-            left = min(self.total_bounds[0], centr.total_bounds[0])
-            bottom = min(self.total_bounds[1], centr.total_bounds[1])
-            right = max(self.total_bounds[2], centr.total_bounds[2])
-            top = max(self.total_bounds[3], centr.total_bounds[3])
-            crs = self.meta['crs']
-            width = (right - left) / self.meta['transform'][0]
-            height = (bottom - top) / self.meta['transform'][4]
-            self.meta = {
-                'dtype': 'float32',
-                'width': width,
-                'height': height,
-                'crs': crs,
-                'transform': rasterio.Affine(self.meta['transform'][0], 0.0, left,
-                                             0.0, self.meta['transform'][4], top),
-            }
-            self.lat, self.lon = np.array([]), np.array([])
+
+        if self.meta and not self.lat.any():
+            self.set_meta_to_lat_lon()
+        if centr.meta and not centr.lat.any():
+            centr.set_meta_to_lat_lon()
+
+        if not u_coord.equal_crs(centr.geometry.crs, self.geometry.crs):
+            raise ValueError('Different CRS not accepted.')
+
+        self.meta = dict()
+        self.__dict__.update(self.union(centr).__dict__)
```

Comment:
I think we should implement `Centroids.union` in such a way that we can reduce this to a single line:
```suggestion
        self.__dict__ = self.union(centr).__dict__
```